src/init.cpp

```diff
@@ -1519,11 +1519,12 @@ bool AppInitMain(NodeContext& node, interfaces::BlockAndHeaderTipInfo* tip_info)
 
     // ********************************************************* Step 10: data directory maintenance
 
-    // if pruning, unset the service bit and perform the initial blockstore prune
-    // after any wallet rescanning has taken place.
-    if (fPruneMode) {
-        LogPrintf("Unsetting NODE_NETWORK on prune mode\n");
-        nLocalServices = ServiceFlags(nLocalServices & ~NODE_NETWORK);
+    if (!fPruneMode) {
```

Comment:
nit: Generally it is recommended if the `if` has two branches to not invert the condition to avoid a double negative in the `else` branch

Reply:
Thanks, fixed that by keeping the original condition and setting the flag in the else-branch instead.